ui/src/app/views/notebook-list/component.ts

```diff
@@ -158,6 +168,64 @@ export class NotebookListComponent implements OnInit, OnDestroy {
     this.openNotebook();
   }
 
+  renameThis(notebook: FileDetail): void {
+    this.notebookInFocus = notebook;
+    this.renameModal.open();
+  }
+
+  receiveRename($event: NotebookRename): void {
+    this.renameNotebook($event);
+  }
+
+  renameNotebook(rename: NotebookRename): void {
+    let newName = rename.newName;
+    if (!(new RegExp('^.+\.ipynb$').test(newName))) {
+      newName = rename.newName + '.ipynb';
+      rename.newName = newName;
+    }
+    if (new RegExp('.*\/.*').test(newName)) {
+      this.renameModal.close();
+      this.notebookRenameError = true;
+      return;
+    }
+    if (this.notebookList.filter((nb) => nb.name === newName).length > 0) {
+      this.renameModal.close();
+      this.notebookRenameConflictError = true;
+      this.duplicateName = newName;
+      return;
+    }
+   this.workspacesService.renameNotebook(this.wsNamespace, this.wsId, rename).subscribe((newNb) => {
+      this.notebookList.splice(
+        this.notebookList.indexOf(this.notebookInFocus), 1, newNb
+      );
+     this.renameModal.close();
+   });
+  }
+
+  cloneThis(notebook: string): void {
+    this.workspacesService.cloneNotebook(this.wsNamespace, this.wsId, notebook)
```

Comment:
We might want this method to return the newly cloned notebook (that would be consistent with the rename). Though in general I also like the approach of refreshing the full list.

Reply:
I thought of "rename" as essentially a "update" so as Bru suggested I returned a new FileDetail. I prefer this for clone.

Reply:
Could you expand "this". Do you mean you prefer what it currently returns (nothing)? To be clear I was suggesting it return a FileDetail. In general if you have a method that creates something it returns that thing.

Anyways, it's not super important since I'm not sure it will be used but it does seem like a better/more consistent API - see CloneWorkspace, CloneCohort which return the new thing.

Reply:
It's not used at all. "this" being how I have it currently set up.

Reply:
But I will change it... That said I feel like this adds a bunch of code, for not a extremely strong reason.